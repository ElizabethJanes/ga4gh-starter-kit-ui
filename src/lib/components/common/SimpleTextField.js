import React from 'react';
import { 
    Typography,
    FormControl,
    TextField,
    Grid,
    Button
} from '@material-ui/core';
import { v4 as uuidv4 } from 'uuid';

const SimpleTextField = props => {
    return(
        <FormControl fullWidth>
            <TextField
                id={props.id}
                label={props.label}
                margin='normal'
                name={props.name}
                type='text' 
                value={props.value}
                InputProps={{readOnly: props.readOnly}}
                onChange={e => props.changeFunction(e.target.value)}
                helperText={props.helperText}/>
        </FormControl>
    );
}

export default SimpleTextField;