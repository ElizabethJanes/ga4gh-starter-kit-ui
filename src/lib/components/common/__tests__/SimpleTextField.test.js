import React from 'react';
import renderer from 'react-test-renderer';
import SimpleTextField from '../SimpleTextField';

test('SHOW <SimpleTextField />', () => {
    const simpleTextField = renderer
    .create(<SimpleTextField 
        readOnly={true} 
        property='snapshot_test' 
        propertyName='Snapshot Test' 
        label='Snapshot Test'
        description='This is a description for a snapshot test.' />)
    .toJSON();
    expect(simpleTextField).toMatchSnapshot();
});

test('NEW and EDIT <SimpleTextField />', () => {
    const simpleTextField = renderer
    .create(<SimpleTextField 
        readOnly={false} 
        property='snapshot_test' 
        propertyName='Snapshot Test' 
        label='Snapshot Test'
        description='This is a description for a snapshot test.' />)
    .toJSON();
    expect(simpleTextField).toMatchSnapshot();
});