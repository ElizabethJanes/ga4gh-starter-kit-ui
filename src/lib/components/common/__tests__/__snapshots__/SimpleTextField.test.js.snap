// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NEW and EDIT <SimpleTextField /> 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      Snapshot Test
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={false}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;

exports[`SHOW <SimpleTextField /> 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth"
>
  <div
    className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
      data-shrink={false}
    >
      Snapshot Test
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiInput-input"
        disabled={false}
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        readOnly={true}
        required={false}
        type="text"
      />
    </div>
  </div>
</div>
`;
